ooniprobe/Model/Database/Measurement.mm

```diff
@@ -96,4 +96,24 @@ -(void)deleteObject{
     [self remove];
 }
 
+- (void)getExplorerUrl:(void (^)(NSString *))measurement_url {
+    NSString *path = [NSString stringWithFormat:@"https://api.ooni.io/api/v1/measurements?report_id=%@&input=%@", self.report_id, self.url_id.url];
+    NSURL *url = [NSURL URLWithString:path];
+    NSURLSessionDataTask *downloadTask = [[NSURLSession sharedSession] dataTaskWithURL:url completionHandler:^(NSData *data, NSURLResponse *response, NSError *error) {
+        if (!error) {
+            NSDictionary *dic = [NSJSONSerialization JSONObjectWithData:data options:0 error:&error];
+            NSArray *resultsArray = [dic objectForKey:@"results"];
+            if ([resultsArray count] == 0)
+                measurement_url(nil);
+            measurement_url([[resultsArray objectAtIndex:0] objectForKey:@"measurement_url"]);
+        }
+        else {
+            // Fail
+            measurement_url(nil);
+            NSLog(@"error : %@", error.description);
+        }
```

Comment:
Please, refactor this code as:

```ObjectiveC
if (error) {
  // handle error
  return
}
// handle normal case
```

There is a bunch of reasons for doing so. Not least that most people expect code to be like that.

Reply:
For what I know calling the callback is a return

Reply:
No, this is not correct. You need to `return` explicitly right after calling the callback. The code inside a block is not special and all the usual code flow rules apply.